### PR TITLE
Get rid of volatile configuration fields for Scheduler

### PR DESCRIPTION
**Pull request by geoand**

### CODE REVIEW

**Comment by geoand** [comment]
Weird that tests are failing... I haven't been able to reproduce the failure locally.

I'll check later on

**Comment by geoand** [comment]
Tests should now be fixed.

I've got to say, `SyntheticBeanBuildItem` is really handy!

File: extensions/quartz/runtime/src/main/java/io/quarkus/quartz/runtime/QuartzScheduler.java [hunk 1]
```diff
@@ -68,13 +67,13 @@
     }
 
     public QuartzScheduler(SchedulerContext context, QuartzSupport quartzSupport, Config config) {
+        Map<String, ScheduledInvoker> invokers;
```
Comment (machi1990) [inline extensions/quartz/runtime/src/main/java/io/quarkus/quartz/runtime/QuartzScheduler.java:70]: Can be moved to where the initialisation occurs? https://github.com/quarkusio/quarkus/pull/9026/files#diff-f3233ae68279acfe2869eaffaeb10682R76

File: extensions/quartz/deployment/src/main/java/io/quarkus/quartz/deployment/QuartzProcessor.java [hunk 2]
```diff
@@ -168,13 +170,22 @@ private String guessDriver(Optional<JdbcDataSourceBuildItem> jdbcDataSource) {
...
     }
+
+    @BuildStep
+    @Record(RUNTIME_INIT)
+    public void quartzSupportBean(QuartzRuntimeConfig runtimeConfig, QuartzBuildTimeConfig buildTimeConfig,
+            QuartzRecorder recorder,
+            BuildProducer<SyntheticBeanBuildItem> syntheticBeanBuildItemBuildProducer,
+            QuartzJDBCDriverDialectBuildItem driverDialect) {
+
+        syntheticBeanBuildItemBuildProducer.produce(SyntheticBeanBuildItem.configure(QuartzSupport.class)
```
Comment (machi1990) [inline extensions/quartz/deployment/src/main/java/io/quarkus/quartz/deployment/QuartzProcessor.java:186]: Ah did not know we have this. Thanks :)

**Review by machi1990 — APPROVED** [review]
Added one minor suggestion, tests are passing. LGTM.

File: extensions/quartz/runtime/src/main/java/io/quarkus/quartz/runtime/QuartzScheduler.java (hunk 1 above)
Comment (geoand) [reply extensions/quartz/runtime/src/main/java/io/quarkus/quartz/runtime/QuartzScheduler.java]: Good idea, done

File: extensions/quartz/deployment/src/main/java/io/quarkus/quartz/deployment/QuartzProcessor.java (hunk 2 above)
Comment (geoand) [reply extensions/quartz/deployment/src/main/java/io/quarkus/quartz/deployment/QuartzProcessor.java:186]: Yeah. @mkouba has improved the experience of "creating" beans at "build time".

**Comment by geoand** [comment]
@mkouba OK with you too?

**Review by mkouba — APPROVED** [review]
LGTM

**Pull request merged**